### ISSUE TITLE
Ability to set Resolver per-module

### ISSUE THREAD

**Issue by trentmwillis** [switchfly/ember-test-helpers#173]
In the current world, it is easy to set a Resolver to be used for an entire test-suite by using the `setResolver` method. This is fine in the vast majority of use cases, but with the advent of Engines, and particularly in-repo-engines, we need some additional flexibility [as shown in this issue](https://github.com/dgeb/ember-engines/issues/123).

One approach that I suggest is to allow setting a `resolver` option that is passed into the test module like so:

```js
import EngineResolver from 'some-engine/resolver';

moduleForComponent('awesome-component', {
  resolver: EngineResolver.create()
});
```

Then, when constructing the container/registry for that test module, we'll use the passed in Resolver instead of the globally defined one that was set with `setResolver`.

This provides backwards compatibility while giving enough flexibility that the ember-engines addon could provide an additional wrapped to allow getting a proper resolver by engine-name. I imagine something like:

```js
moduleForComponent('awesome-component', {
  resolver: getEngineResolver('some-engine')
});
```

**Comment by rwjblue** [comment]
This seems good to me.

**Comment by dgeb** [comment]
Works for me as well 👍

**Comment by Turbo87** [comment]
resolved by #174

**Comment by rwjblue** [comment]
Thank you @turbo87
